modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/RegisteredDomainProcessorTests.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.ingest.common;
+
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public class RegisteredDomainProcessorTests extends ESTestCase {
+    private Map<String, Object> buildEvent(String domain) {
+        return new HashMap<>() {
+            {
+                put("domain", domain);
+            }
+        };
+    }
+
+    public void testBasic() throws Exception {
+        testRegisteredDomainProcessor(buildEvent("www.google.com"), "www.google.com", "google.com", "com", "www");
+        testRegisteredDomainProcessor(buildEvent(""), null, null, null, null);
+        testRegisteredDomainProcessor(buildEvent("."), null, null, null, null);
+        testRegisteredDomainProcessor(buildEvent("$"), null, null, null, null);
+        testRegisteredDomainProcessor(buildEvent("foo.bar.baz"), null, null, null, null);
+        testRegisteredDomainProcessor(
+            buildEvent("1.www.global.ssl.fastly.net"),
+            "1.www.global.ssl.fastly.net",
+            "www.global.ssl.fastly.net",
+            "global.ssl.fastly.net",
+            "1"
+        );
+    }
```

Comment:
These test cases look a little thin. In particular, I would like to see at least one test with a two-part eTLD such as `<something>.co.uk`.